src/textual/widgets/_selection_list.py

```diff
@@ -0,0 +1,655 @@
+            for a message sent when any change (selected or deselected,
+            either by user interaction or by API calls) is made to the
+            selected values.
+        """
+
+    @dataclass
+    class SelectedChanged(Generic[MessageSelectionType], Message):
+        """Message sent when the collection of selected values changes.
+
+        This message is sent when any change to the collection of selected
+        values takes place; either by user interaction or by API calls.
+        """
+
+        selection_list: SelectionList[MessageSelectionType]
+        """The `SelectionList` that sent the message."""
+
+        @property
+        def control(self) -> SelectionList[MessageSelectionType]:
+            """An alias for `selection_list`."""
+            return self.selection_list
+
+    def __init__(
+        self,
+        *selections: Selection
+        | tuple[TextType, SelectionType]
+        | tuple[TextType, SelectionType, bool],
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """Initialise the selection list.
+
+        Args:
+            *content: The content for the selection list.
+            name: The name of the selection list.
+            id: The ID of the selection list in the DOM.
+            classes: The CSS classes of the selection list.
+            disabled: Whether the selection list is disabled or not.
+        """
+        self._selected: dict[SelectionType, None] = {}
+        """Tracking of which values are selected."""
+        self._send_messages = False
+        """Keep track of when we're ready to start sending messages."""
+        super().__init__(
+            *[self._make_selection(selection) for selection in selections],
+            name=name,
+            id=id,
+            classes=classes,
+            disabled=disabled,
+        )
+
+    @property
+    def selected(self) -> list[SelectionType]:
+        """The selected values.
+
+        This is a list of all of the
+        [values][textual.widgets.selection_list.Selection.value] associated
+        with selections in the list that are currently in the selected
+        state.
+        """
+        return list(self._selected.keys())
+
+    def _on_mount(self) -> None:
+        """Configure the list once the DOM is ready."""
+        self._send_messages = True
+
+    def _message_changed(self) -> None:
+        """Post a message that the selected collection has changed, where appropriate.
+
+        Note:
+            A message will only be send if `_send_messages` is `True`. This
+            makes this safe to call before the widget is ready for posting
+            messages.
+        """
+        if self._send_messages:
+            self.post_message(self.SelectedChanged(self))
+
+    def _apply_to_all(self, state_change: Callable[[SelectionType], bool]) -> Self:
```

Comment:
I suggest an explicit note in the docstring saying we get a single message changed regardless of how many selections actually change.